platforms/documentation/docs/src/docs/userguide/best-practices/best_practices_general.adoc

```diff
@@ -175,3 +175,92 @@ include::sample[dir="snippets/bestPractices/avoidInternal/groovy",files="build.g
 === Tags
 
 <<tags_reference.adoc#tag:upgrades,`#upgrades`>>
+
+[[use_the_gradle_properties_file]]
+== Set build flags in `gradle.properties`
+:keywords: properties,structuring-builds
+
+Set Gradle build property flags in the `gradle.properties` file.
+
+=== Explanation
+
+Instead of using command-line options or environment variables, set build flags in the root project's `gradle.properties` file.
+
+Gradle comes with a long list of <<build_environment.adoc#sec:gradle_configuration_properties,Gradle properties>>, which have names that begin with `org.gradle` and can be used to configure the behavior of the build tool.
+You should familiarize yourself with these properties, which can have **enormous** impacts on your build's performance.
+
+You should not rely on supplying these properties on the CLI with every Gradle invocation.
+Providing these properties on the command line is meant for temporary testing and debugging, and is easy to forget across different environments.
+A permanent, idiomatic location to set and share these properties across different machines and between developers is in the `gradle.properties` file located in the root project directory.
```

Comment:
This is too much in one sentence, let's shorten it:
```suggestion
A permanent, idiomatic location to set and share these properties is in the `gradle.properties` file located in the root project directory.
```
Maybe a second sentence here about sharing the file and checking it into a repo?